modules/apps/29-fee/ibc_middleware_test.go

```diff
@@ -1004,6 +1004,90 @@ func (suite *FeeTestSuite) TestOnTimeoutPacket() {
 	}
 }
 
+func (suite *FeeTestSuite) TestOnChanUpgradeInit() {
+	var path *ibctesting.Path
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			"invalid upgrade version",
+			func() {
+				path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = "invalid-version"
+				path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = "invalid-version"
+
+				suite.chainA.GetSimApp().FeeMockModule.IBCApp.OnChanUpgradeInit = func(_ sdk.Context, _, _ string, _ channeltypes.Order, _ []string, _ uint64, _, _ string) (string, error) {
+					return "", ibcmock.MockApplicationCallbackError
```

Comment:
Intentionally force the error here so we can assert that a passthrough results in `isFeeEnabled == false`

Reply:
should we mention this in the test case?

Reply:
yeah definitely, good call!